src/Microsoft.ServiceFabric.Services/Client/ServicePartitionResolver.cs

```diff
@@ -621,15 +621,13 @@ public static ServicePartitionResolver GetDefault()
                             // Registering for Notification only for the first request for a service uri.
                             if (this.UseNotification && !this.registrationCache.ContainsKey(serviceUri))
                             {
-                                var added = this.registrationCache.TryAdd(serviceUri, true);
-                                if (added)
-                                {
-                                    ServiceNotificationFilterDescription filter = new ServiceNotificationFilterDescription(
-                                       name: serviceUri,
-                                       matchNamePrefix: true,
-                                       matchPrimaryChangeOnly: false);
-                                    await client.ServiceManager.RegisterServiceNotificationFilterAsync(filter, totaltime.GetRemainingTime(), CancellationToken.None);
-                                }
+                                ServiceNotificationFilterDescription filter = new ServiceNotificationFilterDescription(

```

Comment:
Why did the order of instructions change? i.e, why TryAdd moved to after registration?

Reply:
Actually I just saw, we are removing the service uri entry from the registrationCache in case the RegisterServiceNotificationFilterAsync fails. So the previous flow would have worked also.
Let me change back to that.